cr.h

```diff
@@ -1103,7 +1103,37 @@ static cr_plugin_main_func cr_so_symbol(so_handle handle) {
     return new_main;
 }
 
+#ifdef __MINGW32__
+#include <setjmp.h>
+#include <signal.h>
+
+static jmp_buf env;
+static void cr_signal_handler(int sig) {
+    __builtin_longjmp(env, 1);
+}
+
+static cr_failure cr_signal_to_failure(int sig) {
+  switch (sig) {
```

Comment:
Overall this looks good to me, unfortunately our CI tests are broken due travis ending support and I did not migrate it to Github Actions, so I'll be going a bit blindly here. :crossed_fingers: 

Note that I would like you to keep the standard formatting in the file (4-spaces instead 2-spaces from your changes), so please fix that too.

Reply:
My bad, I did not applied clang formatting to my code block. 4 spaces were corrected.